1-bootstrap/README.md

```diff
@@ -1,32 +1,381 @@
+
+   To enable the API, execute the following command, replacing `YOUR_SECRET_PROJECT_ID` with your actual project ID:
+
+   ```bash
+   gcloud services enable \
+   secretmanager.googleapis.com \
+   --project=YOUR_SECRET_PROJECT_ID
+   ```
+
+### KMS Project
+
+A separate Google Cloud project is required to store the KMS key by this solution.
+
+- __IAM Roles:__ The identity deploying the module requires the following IAM role on the KMS project:
+
+   - Project IAM Admin: `roles/resourcemanager.projectIamAdmin`
+
+   ```bash
+   gcloud projects add-iam-policy-binding YOUR_KMS_PROJECT_ID \
+   --member="serviceAccount:YOUR_SERVICE_ACCOUNT_EMAIL" \
+   --role="roles/resourcemanager.projectIamAdmin"
+   ```
+
+- __API Enabled:__ The following API must be enabled in the Seed and KMS projects:
+
+   - `cloudkms.googleapis.com`
+
+   To enable the API, execute the following command, replacing `YOUR_KMS_PROJECT_ID` with your actual project ID:
+
+   ```bash
+   gcloud services enable \
+   cloudkms.googleapis.com \
+   --project=YOUR_KMS_PROJECT_ID
+   ```
+
+####  KMS Key for Bucket Encryption
+
+A KMS key will be used to encrypt the contents of the created Cloud Storage buckets. This key should reside in the KMS Project.
+
+####  KMS Key for Binary Authorization Attestation
+
+A KMS key will be used to sign images during building time. This key should reside in the KMS Project.
 
-#### Secrets Project
+### Logging Bucket
 
-You will need a Google Cloud project with [Secret Manager](https://cloud.google.com/security/products/secret-manager) to store your git credentials, throughout the documentation this will be referenced as `$GIT_SECRET_PROJECT`.
+You can optionally specify an existing Cloud Storage bucket to store logs from:
+
+- Build logs
+- Terraform state bucket
+
+The bucket will use the KMS Key provided to encrypt the content. In this case, the code will grant the Storage Service Agent:
+
+   - Cloud KMS CryptoKey Encrypter: `roles/cloudkms.cryptoKeyEncrypter`
+   - Cloud KMS CryptoKey Decrypter: `roles/cloudkms.cryptoKeyDecrypter`
+
+If a Key is not provided, a new one will be created at the same project to encrypt the content.
+
+### VPC Service Controls (VPC-SC)
+
+This module supports deployment within a VPC-SC perimeter.
+
+__Important:__ The Cloud Build project (seed project) __cannot__ be within the perimeter. This is because the module creates new projects, and errors will occur when accessing services (e.g., enabling APIs) before the new project is added to the perimeter.
+
+This module does not create the Service Perimeter or Access Level. However, it can add projects to the Service Perimeter, create directional rules, and add identities to the Access Level.
+
+To enable VPC-SC integration, you must provide the following:
+
+- An existing Access Level name.
+- An existing Service Perimeter name.
+- The deployment mode (`DRY_RUN` or `ENFORCED`).
+
+The identity deploying the module must be a member of the specified Access Level.
```

Comment:
All the examples are using a service account. does it **need** to be a service account?